### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by HeyItsGwen**
Pulling from feature branch to development branch after assumed completion of the project specifications.

    Made posts model/functionality: Titles, contents, user_id, new/edit/destroy functionality, attached to user on creation.
    Made login/out functionality, as well as sign up and edit account options, and added name to the user model to connect with posts.
    Added the 'members only' functionality to make names/created by only appear on posts when logged in.
    Added functionality to only be able to create/edit/destroy posts when logged in, and only edit/destroy posts that were created by you.

Opening this pull request to ask for a code review and to review any stickler errors.

### CODE REVIEW

**Comment by HeyItsGwen** [comment]
image

I have no idea what's going on with stickler, maybe I'm doing something wrong, but rubocop is giving me zero errors. Is it because i didn't add the stickler.yml file when I started the project, and did it at the end instead? I tried quite a few things, and even the automatic pull request from stickler said "there's already a .stickler.yml file located in that repo! you're good to go!" so I really have no idea what to do here.

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  resources :posts
```
Comment (wathigo) [inline config/routes.rb:2]: **[REQUIRED]**
* It creates the application routes using resources :posts, only: [:new, :create, :index] ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from Odin)

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+class PostsController < ApplicationController
```
Comment (wathigo) [inline app/controllers/posts_controller.rb:1]: **[REQUIRED]**
* it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from Odin)

**Comment by HeyItsGwen** [comment]
added the before action to check for account being logged in before allowing CRUD actions.

image

Left this as is because not only does it include and have functionality for the new, create, and index actions, it also includes update and destroy with authentication, and changes the views based on which user is logged in and which user created the post. (i.e. make two accounts, post on one, log in on the other, you won't see the update/destroy option and if you manually enter the link to edit a post it won't let you if you're not the user who created the post.) I don't feel the need to remove working functionality that doesn't impact the project requirements.

**Comment by HeyItsGwen** [comment]
My project has all of the things mentioned in the project specs and by reviewers.

I just also included the rest of the routes and gave them working functionality and authentication. Am I really being asked to remove good, working things from my project just so that you can see "only" in the file? The project specs don't say "must have only these things", it says it wants at least those things, and even says to add more and dress it up if you want to.

Again. those things are in there and fully working, there's just also more on top of it.

**Comment by HeyItsGwen** [comment]
Thank you Raymond!!

**Pull request merged**